### PR TITLE
Master use search groupby in pivot dam (testing some approach)

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: addons/web/static/tests/views/pivot_tests.js [hunk 1]
```diff
@@ -712,6 +712,322 @@ QUnit.module('Views', {
...
+                    </search>
+                `,
+            },
+        });
+
+        // open group by dropdown
+        await cpHelpers.toggleGroupByMenu(pivot);
+        assert.containsN(pivot, '.o_control_panel .o_cp_bottom_right .o_dropdown_menu .o_menu_item', 2,
+            "should have 2 dropdown items in searchview groupby");
+        await testUtils.dom.click(pivot.$('.o_control_panel .o_cp_bottom_right .o_generator_menu button.o_add_custom_group_by'));
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/pivot_tests.js:858]: There are more cpHelpers related to the GroupByMenu that should be use here :-)

**Review by Polymorphe57 — COMMENTED** [review]
Nice job! I have kept your idea but I have reworked a bit the code to avoid some detours:

- almost everything is done directly in the pivot renderer or "below
- the two small classes PivotGroupByMenu and PivotCustomGroupByItem inherit directly from 
 DropdownMenu and DropdownMenuItem

I hope you agree with me that it is better like that.

Could you squash the commits and make a single commit message? Some remark, the group tag in the search arch is just there to separate items visually (groupNumber) and/or in groups (groupId). This is related to the separtors in the control panel dropdowns and to the way filters are combined with AND/OR.

**Comment by Polymorphe57** [comment]
Also something needs to be done in enterprise: we should check if there is not problem in mobile mode. I have created a PR:
https://github.com/odoo/enterprise/pull/17341

File: addons/web/static/tests/views/pivot_tests.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ QUnit.module('Views', {
                     product_id: {string: "Product", type: "many2one", relation: 'product', store: true},
                     other_product_id: {string: "Other Product", type: "many2one", relation: 'product', store: true},
                     non_stored_m2o: {string: "Non Stored M2O", type: "many2one", relation: 'product'},
-                    customer: {string: "Customer", type: "many2one", relation: 'customer', store: true},
+                    customer: {string: "Customer", type: "many2one", relation: 'customer', store: true},
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/pivot_tests.js:40]: remove

**Comment by msh-odoo** [comment]
> Nice job! I have kept your idea but I have reworked a bit the code to avoid some detours:
> 
>     * almost everything is done directly in the pivot renderer or "below
> 
>     * the two small classes PivotGroupByMenu and PivotCustomGroupByItem inherit directly from
>       DropdownMenu and DropdownMenuItem
> 
> 
> I hope you agree with me that it is better like that.
> 
> Could you squash the commits and make a single commit message? Some remark, the group tag in the search arch is just there to separate items visually (groupNumber) and/or in groups (groupId). This is related to the separtors in the control panel dropdowns and to the way filters are combined with AND/OR.

@Polymorphe57 I agree, group tag is for separating items in dropdown, if you check my code I find whether there is groupable items there or not in search by checking searchModel has groupby filters, I think I wrote same description as given in Spec ;)

**Pull request closed without merging**